File: backend/TirthSeva.API/Controllers/DatabaseController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using TirthSeva.API.Data;

namespace TirthSeva.API.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class DatabaseController : ControllerBase
    {
        private readonly ApplicationDbContext _context;

        public DatabaseController(ApplicationDbContext context)
        {
            _context = context;
        }

        [HttpGet("stats")]
        public async Task<ActionResult> GetStats()
        {
            try
            {
                var stats = new
                {
                    TotalUsers = await _context.Users.CountAsync(),
                    VerifiedUsers = await _context.Users.CountAsync(u => u.IsEmailVerified),
                    UnverifiedUsers = await _context.Users.CountAsync(u => !u.IsEmailVerified)
                };

                return Ok(stats);
            }
            catch (Exception ex)
            {
                return BadRequest(new { error = ex.Message });
            }
        }

        [HttpGet("users")]
        public async Task<ActionResult> GetUsers()
        {
            try
            {
                var users = await _context.Users
                    .Select(u => new
                    {
                        u.Id,
                        u.Name,
                        u.Email,
                        u.Role,
                        u.IsEmailVerified,
                        u.EmailOTP,
                        u.OTPExpiry,
                        u.CreatedAt
                    })
                    .ToListAsync();

                return Ok(users);
            }
            catch (Exception ex)
            {
                return BadRequest(new { error = ex.Message });
            }
        }
    }
}

File: backend-spring/src/main/java/com/tirthseva/api/controller/DatabaseController.java
package com.tirthseva.api.controller;

import com.tirthseva.api.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api/database")
@RequiredArgsConstructor
public class DatabaseController {

    private final UserRepository userRepository;

    @GetMapping("/stats")
    public ResponseEntity<?> getStats() {
        try {
            long totalUsers = userRepository.count();
            long verifiedUsers = userRepository.countByIsEmailVerified(true);
            long unverifiedUsers = userRepository.countByIsEmailVerified(false);

            return ResponseEntity.ok(Map.of(
                    "TotalUsers", totalUsers,
                    "VerifiedUsers", verifiedUsers,
                    "UnverifiedUsers", unverifiedUsers));
        } catch (Exception ex) {
            return ResponseEntity.badRequest()
                    .body(Map.of("error", ex.getMessage()));
        }
    }

    @GetMapping("/users")
    public ResponseEntity<?> getUsers() {
        try {
            var users = userRepository.findAll().stream()
                    .map(u -> Map.of(
                            "Id", u.getId(),
                            "Name", u.getName(),
                            "Email", u.getEmail(),
                            "Role", u.getRole(),
                            "IsEmailVerified", u.getIsEmailVerified(),
                            "EmailOTP", u.getEmailOTP() != null ? u.getEmailOTP() : "",
                            "OTPExpiry", u.getOtpExpiry() != null ? u.getOtpExpiry().toString() : "",
                            "CreatedAt", u.getCreatedAt().toString()))
                    .collect(Collectors.toList());

            return ResponseEntity.ok(users);
        } catch (Exception ex) {
            return ResponseEntity.badRequest()
                    .body(Map.of("error", ex.getMessage()));
        }
    }
}
